aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/HttpResponseChecksumDecorator.kt

```diff
@@ -35,7 +35,7 @@ class HttpResponseChecksumDecorator : ClientCodegenDecorator {
 
     // TODO(enableNewSmithyRuntime): Implement checksumming via interceptor and delete this decorator
     private fun applies(codegenContext: ClientCodegenContext, operationShape: OperationShape): Boolean =
-        !codegenContext.settings.codegenConfig.enableNewSmithyRuntime && operationShape.outputShape != ShapeId.from("com.amazonaws.s3#GetObjectOutput")
+        codegenContext.smithyRuntimeMode.generateMiddleware && operationShape.outputShape != ShapeId.from("com.amazonaws.s3#GetObjectOutput")
```

Comment:
Is `generateMiddleware` intended as opposed to `exclusivelyGenerateMiddleware`?

Reply:
I needed to make the request one exclusive since it relies on the middleware property bag, which doesn't exist when generating the serializer for the orchestrator (its customization ends up customizing both serializer implementations). So the exclusivity was to prevent a compiler error when both implementations were enabled.